core/src/main/java/nl/nn/adapterframework/util/JdbcUtil.java

```diff
@@ -872,6 +872,29 @@ public static int executeIntQuery(Connection connection, String query, int param
 		}
 	}
 
+	public static boolean isQueryResultEmpty(Connection connection, String query) throws JdbcException {
+		PreparedStatement stmt = null;
+
+		try {
+			stmt = connection.prepareStatement(query);
+			ResultSet rs = stmt.executeQuery();
+			try {
+				return rs.isAfterLast();
+			} finally {
+				rs.close();
+			}
+		} catch (Exception e) {
+			throw new JdbcException("could not obtain value using query ["+query+"]",e);
+		} finally {
+			if (stmt!=null) {
+				try {
+					stmt.close();
+				} catch (Exception e) {
+					throw new JdbcException("could not close statement of query ["+query+"]",e);
+				}
+			}
+		}
```

Comment:
In plaats van try { } finally, kan je hier een dubbele 'try with resource' doen op statement en op result set:
`		try (PreparedStatement stmt = connection.prepareStatement(query)) {
			try (ResultSet rs = stmt.executeQuery()) {
				return rs.isAfterLast();
			} catch (Exception e) {
				throw new JdbcException("could not obtain value using query ["+query+"]",e);
			}
		} 
`
De finally clauses gaan dan vanzelf.